linode/manager
# test: [M3-8114] - Clean up Linodes, LKE clusters, and Firewalls after Cypress runs

## Description 📝
This adds some post-run resource clean up to our Cypress tests in order to reduce any security risks that come from having Linodes sitting around unattended.

Unfortunately, because of the way that Cypress works and the way our test utils are organized, we can't reuse our existing clean up functions and have to reimplement them (which more or less amounted to a copy/paste). I included some comments about that in the code, and hope to address it long term as part of M3-8803.

## Changes  🔄
- Deletes Linodes, LKE Clusters, and Firewalls after Cypress run

## Target release date 🗓️
N/A

## Preview 📷
![Screenshot 2024-10-30 at 1 31 55 PM](https://github.com/user-attachments/assets/c0348b8e-3cb3-40d1-ac40-23465870eee0)

## How to test 🧪

1. Create some Linodes, Firewalls, and optionally LKE clusters with labels that start with `cy-test-`
  - ℹ️ Note: only LKE clusters that have finished provisioning will be deleted by run clean up, so keep that in mind if you create LKE clusters for the purpose of testing this ticket
2. Run some Cypress test(s) via `yarn cy:run`, example `yarn cy:run -s "cypress/e2e/core/billing/*"`
3. Observe in your console that post-run clean up occurs ("Performing post-run clean up:" note immediately before spec test results are shown)
4. Observe in Cloud Manager that the test resources you created in step 1 have been deleted

## As an Author I have considered 🤔

*Check all that apply*
